examples/servers/simple-auth/mcp_simple_auth/legacy_as_server.py

```diff
@@ -0,0 +1,173 @@
+"""
+Legacy Combined Authorization Server + Resource Server for MCP.
+
+This server implements the old spec where MCP servers could act as both AS and RS.
+Used for backwards compatibility testing with the new split AS/RS architecture.
+
+Usage:
+    python -m mcp_simple_auth.legacy_as_server --port=8002
+"""
+
+import logging
+from typing import Any, Literal
+
+import click
+from pydantic import AnyHttpUrl
+from pydantic_settings import SettingsConfigDict
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import JSONResponse, RedirectResponse, Response
+
+from mcp.server.auth.middleware.auth_context import get_access_token
+from mcp.server.auth.settings import AuthSettings, ClientRegistrationOptions
+from mcp.server.fastmcp.server import FastMCP
+
+from .github_oauth_provider import GitHubOAuthProvider, GitHubOAuthSettings
+
+logger = logging.getLogger(__name__)
+
+
+class ServerSettings(GitHubOAuthSettings):
+    """Settings for the simple GitHub MCP server."""
+
+    model_config = SettingsConfigDict(env_prefix="MCP_")
+
+    # Server settings
+    host: str = "localhost"
+    port: int = 8000
+    server_url: AnyHttpUrl = AnyHttpUrl("http://localhost:8000")
+    github_callback_path: str = "http://localhost:8000/github/callback"
+
+    def __init__(self, **data):
```

Comment:
What is this for? Why is there no type hints?

Reply:
That's the nature of `BaseSettings`, is the idea that they can either be set programmatically or via env var?